cmd/root.go

```diff
@@ -50,11 +66,21 @@ var RootCmd = &cobra.Command{
 backend implementation allowing users to create kubernetes nodes without running the kubelet.
 This allows users to schedule kubernetes workloads on nodes that aren't running Kubernetes.`,
 	Run: func(cmd *cobra.Command, args []string) {
-		f, err := vkubelet.New(nodeName, operatingSystem, kubeNamespace, kubeConfig, provider, providerConfig, taintKey, disableTaint, metricsAddr)
+		ctx := context.Background()
+		f, err := vkubelet.New(ctx, vkubelet.Config{
+			Client:          k8sClient,
+			Namespace:       kubeNamespace,
+			NodeName:        nodeName,
+			Taint:           taint,
+			MetricsAddr:     metricsAddr,
+			Provider:        p,
+			ResourceManager: rm,
+			APIConfig:       apiConfig,
+		})
 		if err != nil {
 			log.L.WithError(err).Fatal("Error initializing virtual kubelet")
 		}
-		if err := f.Run(context.Background()); err != nil {
+		if err := f.Run(ctx); err != nil {
```

Comment:
will it fix #363

Reply:
I don't think so. Wasn't aware of the problem. Will take a look after this.